.github/workflows/main.yml

```diff
@@ -75,79 +134,159 @@ jobs:
           password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
-      # latest-njs-oss image push [Docker Hub]
-      - name: Tag container image for Push to Docker Hub [latest-njs-oss date]
-        run: docker tag nginx-s3-gateway:latest-njs-oss nginxinc/nginx-s3-gateway:latest-njs-oss-${{ steps.date.outputs.date }}
-      - name: Tag container image for Push to Docker Hub [latest-njs-oss]
-        run: docker tag nginx-s3-gateway:latest-njs-oss nginxinc/nginx-s3-gateway:latest-njs-oss
-      - name: Push container image to Docker Hub [latest-njs-oss date]
-        run: docker push nginxinc/nginx-s3-gateway:latest-njs-oss-${{ steps.date.outputs.date }}
-      - name: Push container image to Docker Hub [latest-njs-oss]
-        run: docker push nginxinc/nginx-s3-gateway:latest-njs-oss
-      # unprivileged-oss image push [Docker Hub]
-      - name: Tag container image for Push to Docker Hub [unprivileged-oss date]
-        run: docker tag nginx-s3-gateway:unprivileged-oss nginxinc/nginx-s3-gateway:unprivileged-oss-${{ steps.date.outputs.date }}
-      - name: Tag container image for Push to Docker Hub [unprivileged-oss]
-        run: docker tag nginx-s3-gateway:unprivileged-oss nginxinc/nginx-s3-gateway:unprivileged-oss
-      - name: Push container image to Docker Hub [unprivileged-oss date]
-        run: docker push nginxinc/nginx-s3-gateway:unprivileged-oss-${{ steps.date.outputs.date }}
-      - name: Push container image to Docker Hub [unprivileged-oss]
-        run: docker push nginxinc/nginx-s3-gateway:unprivileged-oss
-      # oss image push [Docker Hub]
-      - name: Tag container image for Push to Docker Hub [oss date]
-        run: docker tag nginx-s3-gateway:oss nginxinc/nginx-s3-gateway:latest-${{ steps.date.outputs.date }}
-      - name: Tag container image for Push to Docker Hub [oss]
-        run: docker tag nginx-s3-gateway:oss nginxinc/nginx-s3-gateway:latest
-      - name: Push container image to Docker Hub [oss date]
-        run: docker push nginxinc/nginx-s3-gateway:latest-${{ steps.date.outputs.date }}
-      - name: Push container image to Docker Hub [oss latest]
-        run: docker push nginxinc/nginx-s3-gateway:latest
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push image [oss]
+        uses: docker/build-push-action@v5
+        with:
+          file: Dockerfile.oss
+          context: .
+          push: true
+          platforms: linux/amd64,linux/arm64
+          provenance: false
+          tags: |
+            ghcr.io/${{ github.repository }}/nginx-oss-s3-gateway:latest-${{ steps.date.outputs.date }}
+            ghcr.io/${{ github.repository }}/nginx-oss-s3-gateway:latest
+            nginxinc/nginx-s3-gateway:latest-${{ steps.date.outputs.date }}
+            nginxinc/nginx-s3-gateway:latest
```

Comment:
**annotation**:
This is nice, the plugin lets you log in to both registries then pushes tot the correct one based on the tag.